content/kb/how-to-migrate-longhorn-chart-installed-in-old-rancher-ui-to-the-chart-in-new-rancher-ui.md

```diff
@@ -0,0 +1,206 @@
+    podsecuritypolicy.policy/longhorn-psp unlabeled
+    rolebinding.rbac.authorization.k8s.io/longhorn-psp-binding annotated
+    rolebinding.rbac.authorization.k8s.io/longhorn-psp-binding unlabeled
+    role.rbac.authorization.k8s.io/longhorn-psp-role annotated
+    role.rbac.authorization.k8s.io/longhorn-psp-role unlabeled
+    serviceaccount/longhorn-service-account annotated
+    serviceaccount/longhorn-service-account unlabeled
+    service/longhorn-backend annotated
+    service/longhorn-backend unlabeled
+    service/longhorn-engine-manager annotated
+    service/longhorn-engine-manager unlabeled
+    service/longhorn-frontend annotated
+    service/longhorn-frontend unlabeled
+    service/longhorn-replica-manager annotated
+    service/longhorn-replica-manager unlabeled
+
+    -----------------------------
+    Successfully updated the annotations and labels for the resources!
+    Next step:
+      1. Go to Rancher UI -> Go to the downstream cluster -> App&Marketplace -> Charts
+      2. Find and select the Longhorn chart
+      3. Select the chart version corresponding the Longhorn version 1.2.3
+      4. Install the chart with the correct helm values. Here are the helm values of your old charts:
+    enablePSP=true,image.defaultImage=true,ingress.enabled=false,longhorn.default_setting=false,persistence.backingImage.enable=false,persistence.defaultClass=true,persistence.defaultClassReplicaCount=3,persistence.reclaimPolicy=Delete,persistence.recurringJobSelector.enable=false,persistence.recurringJobs.enable=false,privateRegistry.registryPasswd=,privateRegistry.registrySecret=,privateRegistry.registryUrl=,privateRegistry.registryUser=,service.ui.type=Rancher-Proxy
+      5. Verify that the migrated charts are working ok
+      6. Run this script again with the flag --type=cleanup to remove the old chart from the legacy UI
+    ```
+4. Go to Rancher UI -> Go to the downstream cluster -> App&Marketplace -> Charts -> Find and select Longhorn chart
+   -> Select the chart version corresponding to the correct Longhorn version
+   -> Install the chart with the correct helm values.
+   Note that all this detailed information is printed out at the end output of the above script.
+5. Verify that the migrated chart (in Rancher UI -> Downstream Cluster -> App & Marketplace -> Installed Apps) is working ok. E.g., check all Longhorn resources inside the namespace `longhorn-system`,
+   try to scale up and down workload to attach/detach Longhorn volumes, ...
+6. At this point the chart is migrated. However, you may notice that the old installation is still visible inside the legacy section:
+   {{< figure src="/img/kb/how-to-migrate-longhorn-chart-installed-in-old-rancher-ui-to-the-chart-in-new-rancher-ui/longhorn-installation-in-legacy-ui.png" >}}
+7. To remove it so you don't accidentally use it in the future, you can run `longhorn_rancher_chart_migration.sh` script with the flag `--type=cleanup`.
```

Comment:
`To remove the legacy icon without affecting Longhorn......`

Reply:
Thank you. Fixed